### PR TITLE
Add possibility to define default params for a single route

### PR DESCRIPTION
**Pull request by vcebotari**
**Motivation**: i want to be able to set default param on a single route, not for entire stack of routes,
for now i can only set 'initialRouteParams' and need this workaround to solve my problem.

**Problem**: i have some categorized list of articles what are in different tabs, want to set header title from route params what i set in default route(tab) params.

**Now it's look like**:
```
Tab1: {
    index: 0,
    params: { title: 'Title 1', url: '/' },
    screen: props => (
      <NewsScreen {...props} />
    ),
 },
Tab2: {
    index: 1,
    params: { title: 'Title 2', url: '/' },
    screen: props => (
      <NewsScreen {...props} />
    ),
 },

```

### CODE REVIEW

**Comment by vcebotari** [comment]
can we refuse to initialRouteParam, and use only params from the route ?

**Comment by brentvatne** [comment]
heya @vcebotari - could we call this `initialParams` instead?

**Comment by ericvicenti** [comment]
imo, `defaultParams` would be a more accurate name for this

File: src/routers/SwitchRouter.js [hunk 1]
```diff
@@ -56,7 +56,11 @@ export default (routeConfigs, config = {}) => {
 
   function resetChildRoute(routeName) {
     const params =
-      routeName === initialRouteName ? initialRouteParams : undefined;
+      routeName === initialRouteName
+        ? initialRouteParams
```
Comment (ericvicenti) [inline src/routers/SwitchRouter.js]: shouldn't this be `{...initalRouteParams, ...routeConfigs[routeName].params }`

**Comment by brentvatne** [comment]
yeah let's do `defaultParams` good call @ericvicenti

File: src/routers/SwitchRouter.js (hunk 1 above)
Comment (vcebotari) [reply src/routers/SwitchRouter.js]: yeah, that's better )

Comment (vcebotari) [reply src/routers/SwitchRouter.js]: @ericvicenti , should we assign initialRouteParams to every child route? or ,
let "initalRouteParams" - for initial route, and simply add defaultParams to child route ?

**Comment by vcebotari** [comment]
@brentvatne , @ericvicenti , if we will use something like: `{...initalRouteParams, ...routeConfigs[routeName].defaultParams }`, maybe we should rename `initialRouteParams` to `defaultStackParams`...

**Comment by brentvatne** [comment]
closing this in favour of: https://github.com/react-navigation/react-navigation-core/pull/11

thanks for getting the ball rolling here!

**Pull request closed without merging**